### PR TITLE
enabling cache dir as a secondary FS mount

### PR DESCRIPTION
**Pull request by cpoole**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md

**What this PR does / why we need it**:
If using the filesystem backend on an NFS mount it is nice to have a second volume mounted for the cache directory since NFS is far too slow for cache. 

This enables an explicit NFS type check in the filestore binarystoreXML and creates a volume for cacheFs

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: stable/artifactory/README.md [hunk 1]
```diff
@@ -119,6 +119,16 @@ To use an NFS server as your cluster's storage, you need to
 ...
 ```
 
+To enable a second volume for cache storage:
```
Comment (eldada) [inline stable/artifactory/README.md]: Can you give some more context to what problem does this feature solve?

File: stable/artifactory/values.yaml [hunk 2]
```diff
@@ -288,6 +289,12 @@ artifactory:
 
     maxCacheSize: 50000000000
     cacheProviderDir: cache
+    cacheFs:
```
Comment (eldada) [inline stable/artifactory/values.yaml]: Missing `cacheFs.persistence.storageClass`

File: stable/artifactory/README.md [hunk 3]
```diff
@@ -618,7 +628,10 @@ The following table lists the configurable parameters of the artifactory chart a
...
 | `artifactory.persistence.customBinarystoreXmlSecret` | A custom Secret for binarystore.xml  | ``                                   |
 | `artifactory.persistence.maxCacheSize`   | The maximum storage allocated for the cache in bytes. | `50000000000`                   |
-| `artifactory.persistence.cacheProviderDir`  | the root folder of binaries for the filestore cache. If the value specified starts with a forward slash ("/") it is considered the fully qualified path to the filestore folder. Otherwise, it is considered relative to the *baseDataDir*. | `cache`                   |
+| `artifactory.persistence.cacheFs.enabled` | Enabled a cache persistent volume | `false` |
+| `artifactory.persistence.cacheFs.persistence.accessMode` | Access mode for the cache persistent volume | `ReadWriteOnce`         |
+| `artifactory.persistence.cacheFs.persistence.bytes`      | Desired size in bytes of the persistent volume | `182536110080`       |
+| `artifactory.persistence.cacheFs.persistence.mountPath`  | Location on the filesystem to mount the cache volume | `/jfrog/cache` |
```
Comment (eldada) [inline stable/artifactory/README.md]: Missing `artifactory.persistence.cacheFs.persistence.storageClass`

File: stable/artifactory/README.md [hunk 4]
```diff
@@ -618,7 +628,10 @@ The following table lists the configurable parameters of the artifactory chart a
 | `artifactory.persistence.binarystoreXml` | Artifactory binarystore.xml template             | See `values.yaml`                    |
 | `artifactory.persistence.customBinarystoreXmlSecret` | A custom Secret for binarystore.xml  | ``                                   |
 | `artifactory.persistence.maxCacheSize`   | The maximum storage allocated for the cache in bytes. | `50000000000`                   |
-| `artifactory.persistence.cacheProviderDir`  | the root folder of binaries for the filestore cache. If the value specified starts with a forward slash ("/") it is considered the fully qualified path to the filestore folder. Otherwise, it is considered relative to the *baseDataDir*. | `cache`                   |
+| `artifactory.persistence.cacheFs.enabled` | Enabled a cache persistent volume | `false` |
+| `artifactory.persistence.cacheFs.persistence.accessMode` | Access mode for the cache persistent volume | `ReadWriteOnce`         |
+| `artifactory.persistence.cacheFs.persistence.bytes`      | Desired size in bytes of the persistent volume | `182536110080`       |
```
Comment (eldada) [inline stable/artifactory/README.md]: Default value you set is `60000000000`

**Comment by eldada** [comment]
@cpoole - can you review my comments?

**Comment by cpoole** [comment]
hey @eldada I addressed your comments. There isn't currently documentation in the readme for the default behavior of the storageclasses so I just put `see yalues.yaml` and put the explanation in values.yaml. 

That seems to be what the chart does in other places as well. 

As for adding to artifactory-ha: I wont have any time this week to look into it and I do not have a license for artifactory HA. Are the automated tests sufficient enough to actually validate the cache directory changes?

File: stable/artifactory/README.md [hunk 5]
```diff
@@ -618,7 +630,11 @@ The following table lists the configurable parameters of the artifactory chart a
 | `artifactory.persistence.binarystoreXml` | Artifactory binarystore.xml template             | See `values.yaml`                    |
 | `artifactory.persistence.customBinarystoreXmlSecret` | A custom Secret for binarystore.xml  | ``                                   |
 | `artifactory.persistence.maxCacheSize`   | The maximum storage allocated for the cache in bytes. | `50000000000`                   |
-| `artifactory.persistence.cacheProviderDir`  | the root folder of binaries for the filestore cache. If the value specified starts with a forward slash ("/") it is considered the fully qualified path to the filestore folder. Otherwise, it is considered relative to the *baseDataDir*. | `cache`                   |
```
Comment (danielezer) [inline stable/artifactory/README.md]: Why did you remove this? I see it is still used.

**Comment by danielezer** [comment]
This was added in PR #563
